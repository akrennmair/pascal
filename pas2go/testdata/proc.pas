program test;

procedure a(procedure b(c : integer); i : integer);
begin
        writeln('foo');
        b(i);
        writeln('bar')
end;

procedure printInt(i : integer);
begin
        writeln('i = ', i)
end;

begin
        a(printInt, 23)
end.
