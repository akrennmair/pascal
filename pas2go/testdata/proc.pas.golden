package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	var a func(b func(int), i int)
	a = func(b func(int), i int) {
		system.Writeln("foo")
		b(i)
		system.Writeln("bar")
		return
	}

	var printint func(i int)
	printint = func(i int) {
		system.Writeln("i = ", i)
		return
	}

	a(printint, 23)
}
